src/components/form/index.tsx

```diff
@@ -76,37 +70,54 @@ export class SmoothlyForm implements Clearable, Submittable, Editable {
 		event.detail(this.readonly)
 	}
 	@Method()
-	async submit(): Promise<void> {
-		this.smoothlyFormSubmit.emit(this.value)
+	async submit(remove?: boolean): Promise<void> {
+		this.processing = true
```

Comment:
move processing inside the if. we have no promise to await here so it will just immediately swap back to false at the ned unless it went inside the if anyway.